pkg/scheduler/apis/config/v1/defaults.go

```diff
@@ -114,17 +118,18 @@ func SetDefaults_KubeSchedulerConfiguration(obj *configv1.KubeSchedulerConfigura
 		obj.Profiles[0].SchedulerName = pointer.StringPtr(v1.DefaultSchedulerName)
 	}
 
-	// Add the default set of plugins and apply the configuration.
-	for i := range obj.Profiles {
-		prof := &obj.Profiles[i]
-		setDefaults_KubeSchedulerProfile(prof)
-	}
-
+	// Set global PercentageOfNodesToScore.
 	if obj.PercentageOfNodesToScore == nil {
 		percentageOfNodesToScore := int32(config.DefaultPercentageOfNodesToScore)
 		obj.PercentageOfNodesToScore = &percentageOfNodesToScore
 	}
 
+	// Add the default set of plugins and apply the configuration.
+	for i := range obj.Profiles {
+		prof := &obj.Profiles[i]
+		setDefaults_KubeSchedulerProfile(prof, obj.PercentageOfNodesToScore)
```

Comment:
In general, it is not good practice to set a default based on another field.

Let's assume you set the global percentage and write the config to a file to use as your baseline for future modifications. Then, you change the global percentage in the file, but since the profiles' percentages where already set, they are not modified.

I would leave the percentages in the profile unmodified. And keep both internal and versioned APIs with pointers, so you can tell if the user tried to override with 0 or they want to inherit the global percentage.

Reply:
> In general, it is not good practice to set a default based on another field.
> 
> Let's assume you set the global percentage and write the config to a file to use as your baseline for future modifications. Then, you change the global percentage in the file, but since the profiles' percentages where already set, they are not modified.
> 
> I would leave the percentages in the profile unmodified. And keep both internal and versioned APIs with pointers, so you can tell if the user tried to override with 0 or they want to inherit the global percentage.

@alculquicondor  IIUIC, scheduler parameter initialization is performed during the scheduler initialization (starts/restarts) and the changes are not persistent. If a user modifies the global percentage, the new value will be propagated to a profile if the profile percentage is not set (empty).  /cc @Huang-Wei

Reply:
It also looks weird to me at first glance, but as @Huang-Wei said, we have default values here, so it looks general ok. 
Another approach how about leave the profile-level `PercentageOfNodesToScore`  a pointer and default to nil, and when getting framework's PercentageOfNodesToScore, it looks like 
```
if profile.PercentageOfNodesToScore == nil {
    return scheduler.PercentageOfNodesToScore
}else{
   return profile.PercentageOfNodesToScore
}
```

The disadvantage is if we have plan to remove the global PercentageOfNodesToScore, although this is still TBD. We'll revert the profile-level PercentageOfNodesToScore to a value obj.

Reply:
> scheduler parameter initialization is performed when a scheduler starts/restarts and a change won't be persistent.

But the scheduler can produce the output into the console. Then, a user can copy-and-paste that for future modification.

> we should have made global PercentageOfNodesToScore a pointer, but given there is no chance to change it now, keeping the profile-level field consistent (as a struct) doesn't sound bad.

That's only the internal struct. We can change it to a pointer, because the versioned one is a pointer.

> The disadvantage is if we have plan to remove the global PercentageOfNodesToScore, although this is still TBD. 

We can't, it's a v1 API now.

Reply:
@alculquicondor Are you suggesting to change profile level percentage to a pointer and not inherit the global value explicitly but return the global value instead if it's nil in `framework.PercentageOfNodesToScore()` as @kerthcet suggested? What about the scheduler level percentage? Shall we keep it as it is or change it to a pointer? Thanks!

Reply:
yes, change both to pointer.

Reply:
Sound good as long as the types are consistent - both struct or both pointers.

Reply:
OK, I'll make the change.

Reply:
Converted it to pointer.